deepchem/data/data_loader.py

```diff
@@ -1613,3 +1616,23 @@ def _featurize_shard(  # type: ignore[override]
             ids.append(entry_id)
         X = np.concatenate(features, axis=0)
         return X, np.array(labels), np.array(weights), np.array(ids)
+
+class DFTYamlLoader(DataLoader):
+    def __init__(self,
+                 featurizer: Featurizer):
+        tasks = None
+    def create_dataset(self, input_files: str, featurizer):
+        entries = self._get_shards(input_files)
+        y = np.array([self._featurize_shard(shard) for shard in entries] )
+        return y 
+        
+    def _get_shards(self, input_files):
+        with open(input_files) as f:
+            data = yaml.load(f, Loader=SafeLoader)
+        return(data)
+    def _featurize_shard(self,
+                         shard):
+        e_type = shard['e_type']
```

Comment:
Make sure to document the required field in the docstring. Also, this should check if these fields are in the Yaml and throw an error if not so there is a good error message for a malformed input